src/k2/module/k23si/Module.cpp

```diff
@@ -284,7 +328,38 @@ void K23SIPartitionModule::_unregisterVerbs() {
     api_server.deregisterAPIObserver("InspectAllKeys");
 }
 
+void K23SIPartitionModule::_registerMetrics() {
+    _metric_groups.clear();
+    std::vector<sm::label_instance> labels;
+    labels.push_back(sm::label_instance("total_cores", seastar::smp::count));
+
+    _metric_groups.add_group("Nodepool", {
+        sm::make_gauge("indexer_keys",[this]{ return _indexer.size();},
+                        sm::description("Number of keys in indexer"), labels),
+        sm::make_counter("read_operations", _readOps, sm::description("Number of read operations"), labels),
+        sm::make_counter("write_operations", _writeOps, sm::description("Number of write operations"), labels),
+        sm::make_counter("query_operations", _queryOps, sm::description("Number of query operations"), labels),
+        sm::make_counter("finalization_operations", _finalizations, sm::description("Number of finalizations requests"), labels),
+        sm::make_counter("push_operations", _pushes, sm::description("Number of pushses"), labels),
+        sm::make_gauge("active_WI", _activeWI, sm::description("Number of active WIs"), labels),
+        sm::make_counter("record_versions", _recordVersions, sm::description("Number of record versions over all records"), labels),
```

Comment:
this should also be a gauge as it can go down (when we remove versions at end of retention window)

Reply:
done